frmts/vrt/vrtderivedrasterband.cpp

```diff
@@ -743,6 +747,73 @@ bool VRTDerivedRasterBand::InitializePython()
     return true;
 }
 
+CPLErr
+VRTDerivedRasterBand::GetPixelFunctionArguments(
+  const CPLString& osMetadata,
+  std::vector<std::pair<CPLString, CPLString>>& oAdditionalArgs)
+{
+
+    CPLXMLNode* oArgs = CPLParseXMLString(osMetadata);
+    if (oArgs != nullptr && oArgs->eType == CXT_Element &&
+        !strncmp(oArgs->pszValue,
+                 "PixelFunctionArgumentsList",
+                 strlen("PixelFunctionArgumentsList")))
+    {
+        for (CPLXMLNode* psIter = oArgs->psChild; psIter != nullptr;
+             psIter = psIter->psNext)
+        {
+            if (psIter->eType == CXT_Element &&
+                !strncmp(psIter->pszValue, "Argument", strlen("Argument")))
+            {
+                CPLString szName, szType, szValue;
+                auto pszName = CPLGetXMLValue(psIter, "name", nullptr);
+                if (pszName != nullptr)
+                    szName = pszName;
+                auto pszType = CPLGetXMLValue(psIter, "type", nullptr);
+                if (pszType != nullptr)
+                    szType = pszType;
+                auto pszValue = CPLGetXMLValue(psIter, "value", nullptr);
+                if (pszValue != nullptr)
+                    szValue = pszValue;
+                if (szType == "constant" && szValue != "" && szName != "")
+                    oAdditionalArgs.push_back(
+                      std::pair<CPLString, CPLString>(szName, szValue));
+                if (szType == "builtin")
+                {
+                    double dfVal;
+                    int success;
+                    if (szValue == "NoData")
+                        dfVal = this->GetNoDataValue(&success);
+                    else if (szValue == "scale")
+                        dfVal = this->GetScale(&success);
+                    else if (szValue == "offset")
+                        dfVal = this->GetOffset(&success);
+                    else
+                    {
+                        CPLError(CE_Failure,
+                                 CPLE_NotSupported,
+                                 "PixelFunction builtin %s not supported",
+                                 szValue.c_str());
+                        return CE_Failure;
+                    }
+                    // Should we signal the user that he is using a builtin
+                    // that depends on a value that is not set on the raster?
+                    // Maybe we should allow generalized use of scale/offset
+                    // even if they are left with their default 1/0 values
+                    oAdditionalArgs.push_back(std::pair<CPLString, CPLString>(
+                      szValue, CPLSPrintf("%lf", dfVal)));
```

Comment:
to get full double accuracy:
```suggestion
                      szValue, CPLSPrintf("%.18g", dfVal)));
```